### ISSUE TITLE
Update Manage Assistants page

### ISSUE THREAD

**Issue by philipperolet** [dust-tt/dust#3266]
As per [here](https://www.figma.com/file/QOxHwppG64GtPocpDhS6Y7/Product?type=design&node-id=4092-63350&mode=design&t=wGVRT1IiZNJ18Mm1-4)

Changes including (but check figma for others)
- The view shows Company assistants but also Shared and Personal
- It allows edition of the gallery
- Search is on all published assistants

Note: the Archived tab is out of scope of this card (other card)
Depends on #3265 

------

Full [figma](https://www.figma.com/file/QOxHwppG64GtPocpDhS6Y7/Product?type=design&node-id=4050-22251&mode=design&t=Dlhq25QyeMahW2OI-0) (source of truth)

Full [issue](https://github.com/dust-tt/decisions/issues/147#issue-2081773564)

**Comment by philipperolet** [comment]
Even clearer specs via the [prototype](https://www.figma.com/proto/QOxHwppG64GtPocpDhS6Y7/Product?type=design&node-id=4050-22252&t=PrquaF04miEgyEMq-0&scaling=min-zoom&page-id=4050%3A22251&starting-point-node-id=4050%3A22252&show-proto-sidebar=1)

**Comment by philipperolet** [comment]
### The Activity Sentence
When we want to explain how much an assistant is used, we use the activity sentence:

"This assistant is active for X members, has sent Y messages in the last 30 days."

- Active for X members means that is in the my assistant list of X members. If it's a shared assistant it means how many people have added it to there list. If it's a company assistant, it means all the members.
- Messages is the number of posted messages by the assistant (not number of threads)

**Comment by philipperolet** [comment]
Note: the "Personal" tab has been discussed, convenience path for builders to update their assistants (cf the issue)

**Comment by philipperolet** [comment]
Note: also cf issue => 'shared' tab uses asisstant preview component, other tabs use lists, as per figma

**Comment by flvndvd** [comment]
@philipperolet what part are we missing from https://github.com/dust-tt/dust/issues/3265 to start this issue?

**Comment by philipperolet** [comment]
cf our slack discussion -> it's the same as #3274 , the dependency is product only. The task can be started, but should not be merged before #3265
